### PR TITLE
COMP: Support reuse of "commit-message" GitHub workflow

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

**Comment by jcfr** [comment]
In the context of the following pull request:
* https://github.com/KitwareMedical/SlicerCustomAppTemplate/pull/62

I was able to confirm the workflow is reusable applying the following change:

```diff
diff --git a/.github/workflows/commit-message.yml b/.github/workflows/commit-message.yml
index 179849d..b4c585b 100644
--- a/.github/workflows/commit-message.yml
+++ b/.github/workflows/commit-message.yml
@@ -15,25 +15,4 @@ permissions:
 
 jobs:
   check-commit-message:
-    name: Check Commit Message
-    runs-on: ubuntu-latest
-    steps:
-      - name: Check Commit Prefix
-        uses: gsactions/commit-message-checker@v2
-        with:
-          pattern: "^(ENH|PERF|BUG|STYLE|DOC|COMP): ([A-Z])+"
-          flags: "gm"
-          excludeDescription: "true" # optional: this excludes the description body of a pull request
-          excludeTitle: "true" # optional: this excludes the title of a pull request
-          error: 'The first line has to start with a commit prefix, followed by a colon and space, and then followed by a message with a capital letter (e.g "ENH: Add support for awesome feature"). For more details on other requirements, see https://slicer.readthedocs.io/en/latest/developer_guide/style_guide.html#commits'
-          checkAllCommitMessages: "true" # optional: this checks all commits associated with a pull request
-          accessToken: ${{ secrets.GITHUB_TOKEN }} # github access token is only required if checkAllCommitMessages is true
-      - name: Check Line Length
-        uses: gsactions/commit-message-checker@v2
-        with:
-          pattern: "^[^#].{1,78}$"
-          error: "The maximum line length of 78 characters is exceeded. For more details, see https://slicer.readthedocs.io/en/latest/developer_guide/style_guide.html#commits"
-          excludeDescription: "true" # optional: this excludes the description body of a pull request
-          excludeTitle: "true" # optional: this excludes the title of a pull request
-          checkAllCommitMessages: "true" # optional: this checks all commits associated with a pull request
-          accessToken: ${{ secrets.GITHUB_TOKEN }} # github access token is only required if checkAllCommitMessages is true
+    uses: jcfr/Slicer/.github/workflows/commit-message.yml@5d187d42f3c834798f9b77d0273b8259f746b91d # support-reuse-commit-message-github-workflow
```

Corresponding workflow execution can be reviewed at https://github.com/KitwareMedical/SlicerCustomAppTemplate/actions/runs/10461483317/job/28969899545?pr=62

As expected it failed because the commit message was purposefully set to an invalid format (i.e. `TEST: Reuse worklflow`)

File: .github/workflows/commit-message.yml [hunk 1]
```diff
@@ -9,6 +9,10 @@ on:
   push:
     branches:
       - main
+  # The "workflow_call" directive allows this workflow to be reused by other workflows.
+  # Example usage in the Slicer Custom Application Template:
+  # https://github.com/KitwareMedical/SlicerCustomAppTemplate/blob/main/.github/workflows/commit-message.yml
```
Comment (jcfr) [inline .github/workflows/commit-message.yml:14]: Note that this URL will be valid only after integrating https://github.com/KitwareMedical/SlicerCustomAppTemplate/pull/62

**Review by lassoan — APPROVED** [review]
Looks good to me, thank you.

File: .github/workflows/commit-message.yml (hunk 1 above)
Comment (jcfr) [reply .github/workflows/commit-message.yml:14]: Also, as describe in https://github.com/Slicer/Slicer/pull/7898#issuecomment-2297525352, we were able to successfully reuse the workflow from the branch associated with `jcfr/Slicer` fork.

**Pull request merged**